src/Leanpkg/Build.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2021 Sebastian Ullrich. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+-/
+import Lean.Data.Name
+import Lean.Elab.Import
+import Leanpkg.Proc
+
+open Lean
+open System
+
+namespace Leanpkg.Build
+
+def buildPath : FilePath := "build"
+def tempBuildPath := buildPath / "temp"
+
+structure Config where
+  pkg      : Name
+  leanArgs : List String
+  leanPath : String
+  -- things like `leanpkg.toml` and olean roots of dependencies that should also trigger rebuilds
+  moreDeps : List FilePath
+
+structure Context extends Config where
+  parents       : List Name := []
+  moreDepsMTime : IO.FS.SystemTime
+
+deriving instance Inhabited for IO.FS.SystemTime
+deriving instance Inhabited for Task
```

Comment:
Should these instances be moved into the Lean core? Since they are not really specific to Leanpkg itself?

Reply:
Yes, they should